### ISSUE TITLE
Hardware emulation

### ISSUE THREAD

**Issue by StrikerX3** [StrikerX3/StrikeBox#1]
- [x] PIC
- [x] PIT (System timer)
- [x] Super I/O (Debug Kits and Dev Kits only)
    - [x] Serial ports
    - [x] Character drivers
        - [x] Null driver
        - [x] Windows serial ports
        - [ ] Linux serial ports
        - [ ] stdin/stdout (low priority)
        - Possibly others, such as TCP, UDP, file and more (low priority)
- [x] CMOS (MC146818 RTC) (#14)
- [x] SMBus
    - [ ] EEPROM: a simple .bin file should do (#7)
    - [x] SMC
        - [x] AV pack
        - [x] Conexant TV encoder
        - [ ] Power/reset controls
        - [x] LED
        - [ ] Fan controller
        - [ ] DVD tray
        - [x] Temperature monitor (ADM1032) (#6)
- [x] PCI bus
  - [x] LPC / ISA bridge
    - [ ] MCPX ROM overlay (#13)
    - [ ] ACPI timer
    - [ ] TV encoder
    - Possibly others
  - [x] PCI bridge
  - [x] AGP bridge
  - [x] IDE controller (#10)
    - [x] ATA/ATAPI-4 / SCSI (#10)
  - [ ] NV2A (#4)
  - [ ] USB (OHCI) (#9)
    - [ ] XID devices (#9)
  - [ ] MCPX APU (#5)
    - [ ] Front-end Engine (FE)
    - [ ] Voice Processor (VP)
    - [ ] Global Processor (GP) (compatible with the Motorola DSP56300)
    - [ ] Encode Processor (EP)
  - [ ] MCPX ACI / AC97 (#5)
  - [ ] NVNet (#8)

Device tree based on [this](https://web.archive.org/web/20180418105344/https://github.com/docbrown/vxb/wiki/Xbox-Hardware-Information), [this](https://web.archive.org/web/20100617022211/http://www.xbox-linux.org/wiki/Xbox_Hardware_Overview) and other sources.

**Comment by StrikerX3** [comment]
I ported over Cxbx-Reloaded's implementation of many devices, with the exception of NV2A for now. It seems that we're hanging at some point during initialization. The system clock should probably attached to  PCI bus and remain disabled until the appropriate command is sent. But there's probably more stuff that needs to be done until we get past the system initialization code and go through the boot animation -- NV2A being an obvious missing component.

**Comment by StrikerX3** [comment]
With #10 solved, StrikeBox can now boot into games.

![image](https://user-images.githubusercontent.com/2067844/49753205-cf205a80-fc99-11e8-94a1-de4e4b201529.png)
